CMakeLists.txt

```diff
@@ -34,6 +34,59 @@ endif()
 
 find_ros1_package(std_msgs REQUIRED)
 
+# ROS1 message packages need to be first in the PYTHONPATH
+# So we remove every ROS2 path from the python path for now
+set(__AMENT_PREFIX_PATH $ENV{AMENT_PREFIX_PATH})
+set(__ORIG_PYTHONPATH $ENV{PYTHONPATH})
+set(__PYTHONPATH "${__ORIG_PYTHONPATH}")
+string(REPLACE ":" ";" __AMENT_PREFIX_PATH ${__AMENT_PREFIX_PATH})
+string(REPLACE ":" ";" __PYTHONPATH ${__PYTHONPATH})
+foreach(ament_path ${__AMENT_PREFIX_PATH})
+  foreach(py_path ${__PYTHONPATH})
+    if(py_path MATCHES "^${ament_path}")
+      list(REMOVE_ITEM __PYTHONPATH ${py_path})
+    endif()
+  endforeach()
+endforeach()
+set(__PYTHONPATH_STRING "")
+foreach(py_path ${__PYTHONPATH})
+  string(CONCAT __PYTHONPATH_STRING "${__PYTHONPATH_STRING}" "${py_path}" ":")
+endforeach()
```

Comment:
Instead of building the Python path in a separate loop could this be populated in the loops above? Also I would suggest to use a CMake list and afterwards replace the semicolons with colons (but not on Windows!).